### DISCUSSION TITLE
Are IDs Automatically Generated in the Suggested CSV Format?

### DISCUSSION THREAD

**Discussion by razakpm** [diabetes-research/help-desk-non-phi#7] [Q&A] [answered]
In the suggested CSV format, are the ID fields (e.g., study_id, participant_id, site_id) automatically generated, or do they need to be manually provided? Clarification is needed on the handling of these ID fields during data ingestion.

**Comment by MeetAnithaVarghese** [comment, marked as answer]
### Unique Identifiers for Each File

Each file contains a unique identifier to distinguish individual records, such as `study_id` for study-related files and `site_id` for site-related files.

- **Study File:** Contains a `study_id` to uniquely identify each study.
- **Site File:** Contains a `site_id` to uniquely identify each site.

> **Note:** The IDs in these files must be manually provided to ensure proper identification of records across different files.

If the provided files don't include the ID entries, they can be generated later in the database by creating a SQL script with the necessary queries, as shown below. However, this approach won't work for foreign keys.


### **ID Generation Sample**
 refer the [SQL](https://raw.githubusercontent.com/surveilr/www.surveilr.com/main/lib/service/diabetes-research-hub/cookbook/sample-id-generation.sql)
```sql
-- Drop the view if it exists
DROP VIEW IF EXISTS drh_participant_data;

-- Create a view with a prefixed and UUID-appended participant_id, and unchanged foreign key values
CREATE VIEW drh_participant_data AS
SELECT
  
  -- Prefix with 'PART_' and append a UUID-like hex blob
  'PART_' || lower(hex(randomblob(2)) || hex(randomblob(2))) AS participant_id,
  
  -- Foreign keys (study_id and site_id) must remain unchanged
  study_id, site_id, diagnosis_icd, med_rxnorm,
  treatment_modality, gender, race_ethnicity, age, bmi, baseline_hba1c,
  diabetes_type, study_arm
FROM uniform_resource_participant;

```
After the automation script is executed, the above SQL can be executed against db and changes can be viewed in UI.
